linera-base/src/identifiers.rs

```diff
@@ -101,6 +99,132 @@ impl FromStr for Account {
     }
 }
 
+impl<'de> Deserialize<'de> for Account {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_struct("Account", &["chain_id", "owner"], AccountVisitor)
+    }
+}
+
+/// A visitor for deserializing [`Account`] instances.
+struct AccountVisitor;
+
+impl<'de> serde::de::Visitor<'de> for AccountVisitor {
+    type Value = Account;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an `Account` represented as a string or an object")
+    }
+
+    fn visit_seq<Accessor>(self, mut accessor: Accessor) -> Result<Self::Value, Accessor::Error>
+    where
+        Accessor: serde::de::SeqAccess<'de>,
+    {
+        let chain_id = accessor.next_element()?.ok_or_else(|| {
+            serde::de::Error::invalid_length(0, &"struct Account with 2 elements")
+        })?;
+
+        let owner = accessor.next_element()?.ok_or_else(|| {
+            serde::de::Error::invalid_length(1, &"struct Account with 2 elements")
+        })?;
+
+        Ok(Account { chain_id, owner })
+    }
+
+    fn visit_map<Accessor>(self, mut accessor: Accessor) -> Result<Self::Value, Accessor::Error>
+    where
+        Accessor: serde::de::MapAccess<'de>,
+    {
+        let mut chain_id = None;
+        let mut owner = None;
+
+        while let Some(key) = accessor.next_key()? {
+            match key {
+                AccountFieldDiscriminator::ChainId => {
+                    ensure!(
+                        chain_id.is_none(),
+                        Accessor::Error::duplicate_field("chain_id")
+                    );
+                    chain_id = Some(accessor.next_value()?);
+                }
+                AccountFieldDiscriminator::Owner => {
+                    ensure!(owner.is_none(), Accessor::Error::duplicate_field("owner"));
+                    owner = Some(accessor.next_value()?);
+                }
+            }
+        }
```

Comment:
alternatively sth like:
```rust
match (accessor.next_key()?, accessor.next_key()?) { ... }
```

Reply:
I'm not sure that's possible, because I think the API requires a call to `next_value` after the `next_key` if you want that entry :thinking: